### ISSUE TITLE
Image in initialMessages when using OpenAI chat causes an error

### ISSUE THREAD

**Issue by devpulse01** [OvidijusParsiunas/deep-chat#28]
Hi Ovidijus,

I've encountered a bug in deep-chat that I hope you can help with.

When initializing the chat with an initialMessages property that includes an image, it seems to cause an error with the OpenAI API. Here's the payload I used:

```json
[
  {
    "text": "Hello!",
    "role": "user"
  },
  {
    "text": "What can you do?",
    "role": "user"
  },
  {
    "files": [
      {
        "src": "https://deepchat.dev/img/city.jpeg",
        "type": "image"
      }
    ],
    "role": "ai"
  },
  {
    "text": "I can assist with a variety of tasks such as answering questions, providing recommendations, translating languages, and much more!",
    "role": "ai"
  }
]
```

Then I got this error:

![initial-messages-error](https://github.com/OvidijusParsiunas/deep-chat/assets/149673277/0501119c-656a-417d-bc41-e27931be8a02)

From the network tab, I can see that OpenAI returns a 400 Bad Request error with the following message:

```json
{
  "error": {
    "message": "'content' is a required property - 'messages.3'",
    "type": "invalid_request_error",
    "param": null,
    "code": null
  }
}
```

And here's what's being sent to OpenAI:

```json
{
    "model": "gpt-4",
    "max_tokens": 2000,
    "temperature": 1,
    "top_p": 1,
    "messages": [
        {
            "role": "system",
            "content": "You are a helpful assistant."
        },
        {
            "content": "Hello!",
            "role": "user"
        },
        {
            "content": "What can you do?",
            "role": "user"
        },
        {
            "role": "assistant"
        },
        {
            "content": "I can assist with a variety of tasks such as answering questions, providing recommendations, translating languages, and much more!",
            "role": "assistant"
        },
        {
            "content": "cool",
            "role": "user"
        }
    ]
}
```

So, it appears that the content property is missing for the message that includes the image. 

Could you please look into this issue? Or maybe I'm doing something wrong on my setup ;)
Thanks in advance for your help!
Best

**Comment by OvidijusParsiunas** [comment]
Hi @devpulse01, thankyou for raising this issue.
The reason why this causes an error is that OpenAI chat is not expected to exchange images, hence when you place an image inside the [`initialMessages`](https://deepchat.dev/docs/messages/#initialMessages) array, the component doesn't process it correctly and sends out a request without a `"content"` property.

I have personally not used GPT-4 yet, but can't find anything relating to images in its API so I would still assume that it does not use images. However please let me know if I am incorrect.

I can patch this issue by simply not including the image object at all, however if the API does expect images, let me know and I'll see what else I can do. Thanks!

**Comment by devpulse01** [comment]
Thank you for the quick response.

Yes, you're absolutely right - the chat API doesn't accept images.
A patch to handle this would be amazing.

Thanks a lot!

**Comment by OvidijusParsiunas** [comment]
The fix for this has been released in version `1.3.23`.

It is very important to note that Deep Chat has many configuration options to allow devs to tailor the chat to their preference. This can become problematic when targeting a specific service that does not allow certain assets. E.g. you can allow user to upload files via the [files](https://deepchat.dev/docs/files) properties and still call the OpenAI chat service which does not allow files - hence resultantly get a service error. This is allowed as they may have a unique use-case for their connection. Hence, as much as we want to provide developers with the freedom to configure and overwrite the chat properties, ultimately they must also be wary of the services they are calling to make sure they have not set up the chat to send something undesirable.

Thanks again for bringing this to my attention, and feel free to raise another issue if you experience any other problems.
